### ISSUE TITLE
Free Weather API

### ISSUE THREAD

**Issue by HotarunIchijou** [breezy-weather/breezy-weather#453]
### Link to publicly available documentation of the API

https://www.weatherapi.com/

### Available and not available data

https://www.weatherapi.com/docs/

### Are you going to develop yourself the new source for the app?

No 

### Give a bit more details or context about this source

I guess it provides the same data as The Weather Channel (weather.com), which uses Google Weather

### Acknowledgements

- [X] This source has a free-tier available (no mandatory credit card info), and if it a main weather source, it includes free hourly forecast
- [X] If this is a weather source request, I checked that endpoints are by coordinates, or that there is a reverse geocoding endpoint.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written the name of the source as the title of this issue.
- [X] I have updated the app to **[version 4.6.5 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)** and this source is not part of it.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
> I guess it provides the same data as The Weather Channel (weather.com), which uses Google Weather

I don't think so, they say "The weather data is provided in partnership with several data providers, government and metreological agencies. " without any more details, but there should be many. Otherwise, we would use the original source directly.

Privacy policy: https://www.weatherapi.com/privacy.aspx

**Comment by papjul** [comment]
The free-tier offers 1 million calls per month (a daily rate-limit would have been better to avoid getting stuck for a few days at the end of the month). This means the source will be in "restricted" caching mode when using the bundled key.

Everything can be queried with a single call. Forecast endpoint includes:
- Current
- Hourly forecast
- Daily forecast
- Alerts
- Air quality

If only current is needed, it may be more efficient to call the dedicated current endpoint.

The search endpoint cannot be used, because it doesn't include timezone.
However, it looks like the timezone endpoint is the same thing with timezone included.
This endpoint can be used to implement a search and reverse geocoding source.
